### PR TITLE
Added standard functor description to functor parameters

### LINKED ISSUE #26737: Add common functor name parameter description
## Reason
Many users do not know what a functor is, so they do not know what can be done with parameters of type `MooseFunctorName`.

## Design
Store a common description string somewhere. Maybe as a public static method in `FunctorInterface`?:
```
static std::string getFunctorDescription() const
{
return "A functor is any of the following: a variable name, a functor material property name, a function name, a post-processor name, or a constant.";
}
```
And then use it like this:
```
params.addRequiredParam<MooseFunctorName>("functor", "The functor to impose. " + getFunctorDescription());
```
Or this instead:
```
params.addRequiredParam<MooseFunctorName>("functor", addFunctorDescription("The functor to impose"));
```
which could be smarter with the ending period/space.

## Impact
Better user experience. Greater usage of functor parameters.

### PR DESCRIPTION
**Pull request by joshuahansel**
Closes #26737

### CODE REVIEW

File: framework/src/utils/InputParameters.C [hunk 1]
```diff
@@ -1049,6 +1057,23 @@ InputParameters::addParam<std::vector<MooseEnum>>(const std::string & /*name*/,
...
+void
+InputParameters::addParam<MooseFunctorName>(const std::string & name,
+                                            const std::string & doc_string)
+{
+  addParamWithoutDefaultInner<MooseFunctorName>(name, appendFunctorDescription(doc_string));
+}
+
+template <>
+void
+InputParameters::addParam<MooseFunctorName, MooseFunctorName>(const std::string & name,
```
Comment (GiudGiud) [inline framework/src/utils/InputParameters.C:1070]: what is the double < T , T > for?

File: framework/include/utils/InputParameters.h [hunk 2]
```diff
@@ -987,6 +987,31 @@ class InputParameters : public Parameters
    */
   void setParameters() {}
 
+  /**
+   * Inner method for \c addRequiredParam
+   */
+  template <typename T>
+  void addRequiredParamInner(const std::string & name, const std::string & doc_string);
```
Comment (GiudGiud) [inline framework/include/utils/InputParameters.h:994]: ugh this is quite verbose for a simple thing. 

Can you try using std::same_if and see if that works?

File: framework/src/utils/InputParameters.C (hunk 1 above)
Comment (joshuahansel) [reply framework/src/utils/InputParameters.C]: There are two template arguments for the `addParam` method that has a default argument, but I guess I don't need the second one (maybe it's deduced implicitly?). Anyway, moot point now with the new changes.

File: framework/include/utils/InputParameters.h (hunk 2 above)
Comment (joshuahansel) [reply framework/include/utils/InputParameters.h]: Works

**Review by GiudGiud — COMMENTED** [review]
Squash commits to remove back & forth (or delete old commits) then good to go imo

We should wait for a test build to look at how it renders in the doco

**Review by GiudGiud — APPROVED** [review]
looks good on doco

**Pull request merged**